// @testing-library/react-native/build/pure.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "act", {
  enumerable: true,
  get: function () {
    return _act.default;
  }
});
Object.defineProperty(exports, "cleanup", {
  enumerable: true,
  get: function () {
    return _cleanup.default;
  }
});
Object.defineProperty(exports, "configure", {
  enumerable: true,
  get: function () {
    return _config.configure;
  }
});
Object.defineProperty(exports, "fireEvent", {
  enumerable: true,
  get: function () {
    return _fireEvent.default;
  }
});
Object.defineProperty(exports, "getDefaultNormalizer", {
  enumerable: true,
  get: function () {
    return _matches.getDefaultNormalizer;
  }
});
Object.defineProperty(exports, "getQueriesForElement", {
  enumerable: true,
  get: function () {
    return _within.getQueriesForElement;
  }
});
Object.defineProperty(exports, "isHiddenFromAccessibility", {
  enumerable: true,
  get: function () {
    return _accessiblity.isHiddenFromAccessibility;
  }
});
Object.defineProperty(exports, "isInaccessible", {
  enumerable: true,
  get: function () {
    return _accessiblity.isInaccessible;
  }
});
Object.defineProperty(exports, "render", {
  enumerable: true,
  get: function () {
    return _render.default;
  }
});
Object.defineProperty(exports, "renderHook", {
  enumerable: true,
  get: function () {
    return _renderHook.renderHook;
  }
});
Object.defineProperty(exports, "resetToDefaults", {
  enumerable: true,
  get: function () {
    return _config.resetToDefaults;
  }
});
Object.defineProperty(exports, "screen", {
  enumerable: true,
  get: function () {
    return _screen.screen;
  }
});
Object.defineProperty(exports, "userEvent", {
  enumerable: true,
  get: function () {
    return _userEvent.userEvent;
  }
});
Object.defineProperty(exports, "waitFor", {
  enumerable: true,
  get: function () {
    return _waitFor.default;
  }
});
Object.defineProperty(exports, "waitForElementToBeRemoved", {
  enumerable: true,
  get: function () {
    return _waitForElementToBeRemoved.default;
  }
});
Object.defineProperty(exports, "within", {
  enumerable: true,
  get: function () {
    return _within.within;
  }
});
var _act = _interopRequireDefault(require("./act"));
var _cleanup = _interopRequireDefault(require("./cleanup"));
var _fireEvent = _interopRequireDefault(require("./fire-event"));
var _render = _interopRequireDefault(require("./render"));
var _waitFor = _interopRequireDefault(require("./wait-for"));
var _waitForElementToBeRemoved = _interopRequireDefault(require("./wait-for-element-to-be-removed"));
var _within = require("./within");
var _config = require("./config");
var _accessiblity = require("./helpers/accessiblity");
var _matches = require("./matches");
var _renderHook = require("./render-hook");
var _screen = require("./screen");
var _userEvent = require("./user-event");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
