// react-native-testing-library/build/helpers/findByAPI.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.findByAPI = exports.findAllByDisplayValue = exports.findByDisplayValue = exports.findAllByPlaceholder = exports.findByPlaceholder = exports.findAllByText = exports.findByText = exports.findAllByTestId = exports.findByTestId = void 0;

var _waitFor = _interopRequireDefault(require("../waitFor"));

var _getByAPI = require("./getByAPI");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const makeFindQuery = (instance, getQuery, text, waitForOptions) => (0, _waitFor.default)(() => getQuery(instance)(text), waitForOptions);

const findByTestId = instance => (testId, waitForOptions = {}) => makeFindQuery(instance, _getByAPI.getByTestId, testId, waitForOptions);

exports.findByTestId = findByTestId;

const findAllByTestId = instance => (testId, waitForOptions = {}) => makeFindQuery(instance, _getByAPI.getAllByTestId, testId, waitForOptions);

exports.findAllByTestId = findAllByTestId;

const findByText = instance => (text, waitForOptions = {}) => makeFindQuery(instance, _getByAPI.getByText, text, waitForOptions);

exports.findByText = findByText;

const findAllByText = instance => (text, waitForOptions = {}) => makeFindQuery(instance, _getByAPI.getAllByText, text, waitForOptions);

exports.findAllByText = findAllByText;

const findByPlaceholder = instance => (placeholder, waitForOptions = {}) => makeFindQuery(instance, _getByAPI.getByPlaceholder, placeholder, waitForOptions);

exports.findByPlaceholder = findByPlaceholder;

const findAllByPlaceholder = instance => (placeholder, waitForOptions = {}) => makeFindQuery(instance, _getByAPI.getAllByPlaceholder, placeholder, waitForOptions);

exports.findAllByPlaceholder = findAllByPlaceholder;

const findByDisplayValue = instance => (value, waitForOptions = {}) => makeFindQuery(instance, _getByAPI.getByDisplayValue, value, waitForOptions);

exports.findByDisplayValue = findByDisplayValue;

const findAllByDisplayValue = instance => (value, waitForOptions = {}) => makeFindQuery(instance, _getByAPI.getAllByDisplayValue, value, waitForOptions);

exports.findAllByDisplayValue = findAllByDisplayValue;

const findByAPI = instance => ({
  findByTestId: findByTestId(instance),
  findByText: findByText(instance),
  findByPlaceholder: findByPlaceholder(instance),
  findByDisplayValue: findByDisplayValue(instance),
  findAllByTestId: findAllByTestId(instance),
  findAllByText: findAllByText(instance),
  findAllByPlaceholder: findAllByPlaceholder(instance),
  findAllByDisplayValue: findAllByDisplayValue(instance)
});

exports.findByAPI = findByAPI;
