// @testing-library/react-native/build/matchers/to-be-disabled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.toBeDisabled = toBeDisabled;
exports.toBeEnabled = toBeEnabled;
var _jestMatcherUtils = require("jest-matcher-utils");
var _hostComponentNames = require("../helpers/host-component-names");
var _textInput = require("../helpers/text-input");
var _componentTree = require("../helpers/component-tree");
var _utils = require("./utils");
function toBeDisabled(element) {
  (0, _utils.checkHostElement)(element, toBeDisabled, this);
  const isDisabled = isElementDisabled(element) || isAncestorDisabled(element);
  return {
    pass: isDisabled,
    message: () => {
      const is = this.isNot ? 'is' : 'is not';
      return [(0, _jestMatcherUtils.matcherHint)(`${this.isNot ? '.not' : ''}.toBeDisabled`, 'element', ''), '', `Received element ${is} disabled:`, (0, _utils.formatElement)(element)].join('\n');
    }
  };
}
function toBeEnabled(element) {
  (0, _utils.checkHostElement)(element, toBeEnabled, this);
  const isEnabled = !isElementDisabled(element) && !isAncestorDisabled(element);
  return {
    pass: isEnabled,
    message: () => {
      const is = this.isNot ? 'is' : 'is not';
      return [(0, _jestMatcherUtils.matcherHint)(`${this.isNot ? '.not' : ''}.toBeEnabled`, 'element', ''), '', `Received element ${is} enabled:`, (0, _utils.formatElement)(element)].join('\n');
    }
  };
}
function isElementDisabled(element) {
  if ((0, _hostComponentNames.isHostTextInput)(element) && !(0, _textInput.isTextInputEditable)(element)) {
    return true;
  }
  const {
    accessibilityState,
    'aria-disabled': ariaDisabled
  } = element.props;
  return ariaDisabled ?? accessibilityState?.disabled ?? false;
}
function isAncestorDisabled(element) {
  const parent = (0, _componentTree.getHostParent)(element);
  if (parent == null) {
    return false;
  }
  return isElementDisabled(parent) || isAncestorDisabled(parent);
}
