// @testing-library/react-native/build/helpers/matchers/match-accessibility-state.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.matchAccessibilityState = matchAccessibilityState;
var _accessiblity = require("../accessiblity");
// This type is the same as AccessibilityState from `react-native` package
// It is re-declared here due to issues with migration from `@types/react-native` to
// built in `react-native` types.
// See: https://github.com/callstack/react-native-testing-library/issues/1351

/**
 * Default accessibility state values based on experiments using accessibility
 * inspector/screen reader on iOS and Android.
 *
 * @see https://github.com/callstack/react-native-testing-library/wiki/Accessibility:-State
 */
const defaultState = {
  disabled: false,
  selected: false,
  checked: undefined,
  busy: false,
  expanded: undefined
};
function matchAccessibilityState(node, matcher) {
  const state = (0, _accessiblity.getAccessibilityState)(node);
  return _accessiblity.accessibilityStateKeys.every(key => matchState(matcher, state, key));
}
function matchState(matcher, state, key) {
  return matcher[key] === undefined || matcher[key] === (state?.[key] ?? defaultState[key]);
}
